### ISSUE TITLE
Update CA1024 to not report on get method return tasks

### ISSUE THREAD

**Issue by Evangelink** [dotnet/roslyn-analyzers#3877]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v3.3.0-beta1.final (Latest pre-release)

#### Diagnostic ID

Example: [CA1024](https://docs.microsoft.com/visualstudio/code-quality/ca1024)

#### Repro steps

Over simplified repro:

```csharp
using System.Threading.Tasks;
public class Something
{
    public Task GetContent()
    {
        return null;
    }
}
```

#### Expected behavior

No diagnostic reported. The docs states that 

> Properties should behave as if they are fields; if the method cannot, it should not be changed to a property.

having `Task`/`ValueTask` return type usually imply there will be some async/await going-on and I don't think it's a good practice for a property.

#### Actual behavior

Diagnostic is reported.

**Comment by Evangelink** [comment]
@mavasani Any opinion on this?

**Comment by mavasani** [comment]
Sounds reasonable change!
